streaming/dataset.py

```diff
@@ -174,98 +174,112 @@ class StreamingDataset(Array, IterableDataset):
+            ``remote``/``local``. Defaults to ``None``.
+        remote (str, optional): Remote path to stream the dataset from. If ``None``, dataset must
+            be complete locally. Defaults to ``None``.
+        local (str, optional): Local working directory to stream the dataset to. Uses a
+            deterministically-calculated temp directory if not set. Defaults to ``None``.
+        split (str, optional): Which dataset split sub-path to use, if any. Defaults to ``None``.
+        index_size (str | int, optional): Expected index size in bytes. Defaults to ``None``.
+        index_hashes (Dict[str, str], optional): Available index hashes. This is a mapping of hash
+            algo name to expected hex digest. Defaults to ``None``.
+        allow_schema_mismatch (bool): If ``True``, continue if schemas mismatch across
+            shards, streams, or the whole dataset. If ``False``, raises if schemas mismatch.
+            Defaults to ``False``.
+        allow_unsafe_types (bool): If ``True``, continue if unsafe type(s) are encountered
+            in shard(s). If ``False``, raises if unsafe type(s) encountered. Defaults to ``False``.
+        allow_unchecked_resumption (bool): If ``True``, upon resume, accept and use shard
+            file phases that we are unable to check the size/hash(es) of. If ``False``, upon
+            resume, drop such files, to regenerate on the fly when needed. Defaults to ``True``.
+        download_retry (str | int): Number of download re-attempts before raising an error.
+            Defaults to ``2``.
+        download_timeout (str | float, optional): Time to wait for a file download to complete
+            before raising an error, if any. Set to ``None`` for no limit. Streaming duration
+            shorthand (e.g., ``1m23s``) is also accepted. Defaults to ``2m``.
+        download_max_size (str | int, optional): Maximum size of an individual download, if any.
+            This is used to prevent over-large shard files from cripping Streaming performance. Set
+            to ``None`` for no size limit. Set to ``str`` to specify the limit using Streaming
+            bytes shorthand. Set to ``int`` to specify bytes. Defaults to ``200mb``.
+        validate_hash (str | Sequence[str], optional): Ranked list of hashing algorithms to
+            apply if expected digest is available. Defaults to ``None``.
+        keep_phases (str | Sequence[str] | Dict[str, Optional[bool]] | Phaser): Which phases
+            to keep and to drop upon conversion, given either by intended use case or literally.
+            Specified as a single use or phase to keep, a sequence of uses or phases to keep, a
+            mapping of uses or phases to whether to keep or drop, or a ``Phaser`` (which performs
+            the same keeping or dropping). Defaults to ``None``.
+        index_download_procs (int, optional): Size of the process pool. You may set either this
```

Comment:
Based on our discussion, you will be removing the `index_download_procs,` `index_download_procs_per_cpu,` and `index_download_max_procs` from here and let SD infer it under the hood when you create the future PRs, right?

Reply:
No, I said these args will be folded into the arg handling for Parquet dataset indexing in the upcoming PR, and there are different possibilities for how to do this correctly. We will not be simply dropping this functionality

Reply:
Also, unless overridden, the parallelism is already inferred in this PR, but by python multiprocessing, up to the limit on concurrent connections

Reply:
Are these args user-customizable? Can we not set defaults based on the instance that the user uses? I am not convinced that we need 3 user-facing args just to tackle the index.json file.

Reply:
Removed x3 per discussion